clang/lib/Sema/SemaType.cpp

```diff
@@ -2305,10 +2305,21 @@ QualType Sema::BuildBitIntType(bool IsUnsigned, Expr *BitWidth,
     return QualType();
   }
 
-  const TargetInfo &TI = getASTContext().getTargetInfo();
-  if (NumBits > TI.getMaxBitIntWidth()) {
+  size_t MaxBitIntWidth = getASTContext().getTargetInfo().getMaxBitIntWidth();
```

Comment:
I wonder if instead we should be setting the aux-target in Host Mode to whatever the device triple is, and looking it up there?  That would generalize this entire thing to : "if NumBits >TI.getMaxBitintWidth() && NumBits > AuxTI.getMaxBitIntWidth()"

Reply:
Are you okay with this being done in a follow-up patch?

Reply:
And should that change be happening upstream as well?

Reply:
I'm a touch concerned as to the behavior changes we might get out of making this a part of the aux target (that is, accepting this and then doing the aux-target changes resulting in different behavior?).  So I would at least want to see that patch be 'close' before accepting this.  

@AaronBallman : While I think that is a GOOD idea (to set  the aux target for host mode upstream), it doesn't help here since upstream doesn't have FPGA.